ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/AbstractSequentialTransactionsLayer.java

```diff
@@ -141,6 +141,7 @@ protected boolean hasExpectedNonce(
   }
 
   @Override
+  @SuppressWarnings("NonApiType")
   protected void internalConsistencyCheck(
       final Map<Address, TreeMap<Long, PendingTransaction>> prevLayerTxsBySender) {
```

Comment:
I think the TreeMap can be changed to a NavigableMap. I think the method we use on the Map is computeIfPresent which is also present on the NavigableMap. I changed this locally and for types that inherit this method and seemed to compile for me.